### DISCUSSION TITLE
Error Handling

### DISCUSSION THREAD

**Discussion by shamxeed** [TanStack/query#2036] [Q&A] [answered]
I have searched the entire docs but haven't seen some guides on error handling and I'm facing a bit of an issue with that.
 On error, I want to get just the error message logged to the console, but getting the entire error log instead.

What I tried: 
```js
const { mutate, status, data, error } = login()

const onLogin = () => {
   if(!username || !password) return
   
   mutate({username, password})
}

useEffect(() => {
   if(status === 'error') {
      console.log(error.message)
   }
}, [status])
```
I want to get just the error message printed to the console but the entire error log is been printed instead which is not what I want because I want to display the error message to the user. THANKS IN ADVANCE!!

**Comment by TkDodo** [comment]
react-query logs all errors to the console by default. You can change that with a custom logger: https://react-query.tanstack.com/reference/setLogger

**Reply by TkDodo** [reply]
For additional error handling, I would recommend the  `onError` callback rather that useEffect

**Comment by shamxeed** [comment, marked as answer]
I archived that by following the error log path to the message.

My use-case:
```js
if(status === 'error') {
   console.log(error.response.errors[0].message)
}
```
